### DISCUSSION TITLE
Query many to many relationship in slqalchemy 2.0

### DISCUSSION THREAD

**Discussion by tomascbk** [sqlalchemy/sqlalchemy#11089] [Usage Questions] [answered]
I'm learning flask with sqlalchemy 2.0 syntax. I would like to query all the tags for a single post. I have the following association table and models. 
[I used this doc to create the many to many table. ](https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#setting-bi-directional-many-to-many)
[This doc to run the Select query](https://docs.sqlalchemy.org/en/20/orm/queryguide/select.html#chaining-multiple-joins)

models.py
```
tags = sa.Table(
    'tags',
    db.metadata,
    sa.Column('post_id', sa.Integer, sa.ForeignKey('post.id'), primary_key=True),
    sa.Column('tag_id', sa.Integer, sa.ForeignKey('tag.id'), primary_key=True)
)

class Post(db.Model):
    __tablename__ = 'post'
    id: so.Mapped[int] = so.mapped_column(primary_key=True)
    body: so.Mapped[str] = so.mapped_column(sa.Text)
    author: so.Mapped[User] = so.relationship(back_populates='posts')
    tags: so.Mapped[List['Tag']] = so.relationship(secondary=tags, back_populates='posts', cascade='all, delete')

class Tag(db.Model):
    __tablename__ = 'tag'
    id: so.Mapped[int] = so.mapped_column(primary_key=True)
    name: so.Mapped[str] = so.mapped_column(sa.String(100))
    posts: so.Mapped[List['Post']] = so.relationship(secondary=tags, back_populates='tags', passive_deletes=True)
```

routes.py
```
@bp.route('/post/<int:id>', methods=['GET', 'POST'])
def post(id):
    post = db.first_or_404(sa.select(Post).where(Post.id == id))
    tags = db.session(sa.select(Tag).join(post.tags).where(Tag.post_id == post.tag_id))
```

Error
`sqlalchemy.exc.ArgumentError: Join target, typically a FROM expression, or ORM relationship attribute expected, got [].`

**Comment by CaselIT** [comment, marked as answer]
Hi,

`tags` is already a relationship so when you access it at the instance level it's already a list of tags, you don't need to perform another query to load them.
I would suggest to take a look at the sqlalchemy tutorial to understand the basis of the ORM https://docs.sqlalchemy.org/en/20/orm/index.html
